### DISCUSSION TITLE
How can we make the staking process more dynamic to prevent passive behavior from contributors?

### DISCUSSION THREAD

**Discussion by xquery-go** [xquery-go/pallet-staking#10] [Q&A]
staking logic

**Comment by slycompiler** [comment]
`One improvement could involve introducing a feature that allows stakers to delegate their stake dynamically based on real-time performance metrics. This would allow contributors to automatically adjust their stake based on the candidate's uptime, reputation, or other performance factors without manually re-staking.`
